tmol/tests/score/common/test_energy_term.py

```diff
@@ -0,0 +1,393 @@
+
+        assert_allclose(gold_vals, scores, atol, rtol)
+
+    @classmethod
+    def test_whole_pose_scoring_gradcheck(
+        cls,
+        pdb,
+        default_database,
+        torch_device,
+        resnums=None,
+        eps=1e-6,  # torch default
+        atol=1e-5,  # torch default
+        rtol=1e-3,  # torch default
+        nondet_tol=0.0,  # torch default
+    ):
+        p1 = pose_stack_from_pdb_and_resnums(pdb, torch_device, resnums)
+
+        whole_pose_scorer = cls.get_pose_scorer(p1, default_database, torch_device)
+
+        def score(coords):
+            scores = whole_pose_scorer(coords)
+            return torch.sum(scores)
+
+        # monkeypatch more sane error reporting
+        torchgrad = importlib.import_module("torch.autograd.gradcheck")
+        torchgrad._get_notallclose_msg = functools.partial(
+            _get_notallclose_msg, atol=atol, rtol=rtol
+        )
+
+        torchgrad.gradcheck(
+            score,
+            (p1.coords.double().requires_grad_(True),),
+            eps=eps,
+            atol=atol,
+            rtol=rtol,
+            nondet_tol=nondet_tol,
+        )
+
+    @classmethod
+    def test_whole_pose_scoring_jagged(
+        cls,
+        pdb,
+        default_database,
+        torch_device: torch.device,
+        update_baseline=False,
+        atol=1e-5,
+        rtol=1e-3,
+    ):
+        res_50 = [(0, 50)]
+        res_30 = [(0, 30)]
+        p1 = pose_stack_from_pdb_and_resnums(pdb, torch_device)
+        p2 = pose_stack_from_pdb_and_resnums(pdb, torch_device, res_50)
+        p3 = pose_stack_from_pdb_and_resnums(pdb, torch_device, res_30)
+        pn = PoseStackBuilder.from_poses([p1, p2, p3], device=torch_device)
+
+        pose_scorer = cls.get_pose_scorer(pn, default_database, torch_device)
+        scores = pose_scorer(pn.coords).cpu().detach().numpy()
+
+        if update_baseline:
+            cls.save_test_baseline_data(
+                cls.test_whole_pose_scoring_jagged.__name__,
+                cls.whole_pose_to_dict(scores),
+            )
+        gold_vals = cls.get_test_baseline_data(
+            cls.test_whole_pose_scoring_jagged.__name__
+        )
+
+        assert_allclose(gold_vals, scores, atol, rtol)
+
+    @classmethod
+    def test_block_scoring_matches_whole_pose_scoring(
+        cls, pdb, default_database, torch_device, resnums=None, atol=1e-5, rtol=1e-3
+    ):
+        p1 = pose_stack_from_pdb_and_resnums(pdb, torch_device, resnums)
+
+        pose_scorer = cls.get_pose_scorer(p1, default_database, torch_device)
+
+        coords = torch.nn.Parameter(p1.coords.clone())
+        block_pair_scores = (
+            pose_scorer(coords, output_block_pair_energies=True).cpu().detach().numpy()
+        )
+        full_pose_scores = (
+            pose_scorer(coords, output_block_pair_energies=False).cpu().detach().numpy()
+        )
+
+        assert_allclose(full_pose_scores, block_pair_scores.sum((2, 3)), atol, rtol)
```

Comment:
Opted to make this a separate test from test_block_scoring since it's clearer and this way we can run it on the full pose.